rust/headless-client/src/ipc_service.rs

```diff
@@ -359,12 +359,14 @@ impl<'a> Handler<'a> {
                 None => Poll::Ready(Event::IpcDisconnected),
             };
         }
-        // `tokio::sync::mpsc::Receiver::recv` is cancel-safe.
-        if let Poll::Ready(option) = self.cb_rx.poll_recv(cx) {
-            return match option {
-                Some(x) => Poll::Ready(Event::Callback(x)),
-                None => Poll::Ready(Event::CallbackChannelClosed),
-            };
+        if let Some(session) = self.session.as_mut() {
+            // `tokio::sync::mpsc::Receiver::recv` is cancel-safe.
+            if let Poll::Ready(option) = session.cb_rx.poll_recv(cx) {
+                return match option {
+                    Some(x) => Poll::Ready(Event::Callback(x)),
+                    None => Poll::Ready(Event::CallbackChannelClosed),
+                };
```

Comment:
This `match` could be a `map` and `unwrap_or` if you fancy using combinators. Not sure if it is more readable.

Reply:
Also, it can sometimes turn out nicer to not use `if let` if you need to nest it more and instead have a `match` that nests the patterns:

```rust
match session.cb_rx.poll_recv(cx) {
	Poll::Ready(Some(x)) => Poll::Ready(Event::Callback(x)),
	// ...
}
```